vm/src/value.rs

```diff
@@ -84,24 +84,226 @@ impl<'vm, ENV> Iterator for PairIter<'vm, ENV> {
     }
 }
 
-// Do this wrap nonsense so that Value is hashable...
-#[derive(Copy, Clone, Debug)]
-pub struct F32Wrap(pub f32);
-
-impl PartialEq for F32Wrap {
+/**
+ * F56 struct
+ * This struct uses 7 bytes to represent a floating point number.
+ * Most operations are done by converting to f64, performing the operation, and then converting back to F56
+ *
+ * F32 uses 1 bit for the sign,  8 bits for the exponent, and 23 bits for the mantissa.
+ * F56 uses 1 bit for the sign, 10 bits for the exponent, and 45 bits for the mantissa.
+ * F64 uses 1 bit for the sign, 11 bits for the exponent, and 52 bits for the mantissa.
+ *
+ *   Byte 0    Byte 1    Byte 2    Byte 3    Byte 4    Byte 5    Byte 6
+ * [sEEEEEEE][EEEmmmmm][mmmmmmmm][mmmmmmmm][mmmmmmmm][mmmmmmmm][mmmmmmmm]
+ *
+ * Exponent bits range from 0 to 1023
+ * they represent -511 to +512 but are stored biased by +511
+ * the exponent of -511 is reserved for the number 0 and subnormal numbers
+ * the exponent of +512 is reserved for infinity and NaN
+ * so normal exponents range from -510 to +511
+ *
+ * smallest positive subnormal value is 8.48e-168 (2^-555)
+ * smallest positive normal value is 2.98e-154 (2^-510)
+ * maximum finite value is 1.34e154
+*/
```

Comment:
turn this into a `///` doc comment. Documnetation comments are additionally useful because
1. They show up in the docs when you build them with `cargo docs`, rust builds documentation for crates automatically.
2. They can have tests: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html